File: frontend/src/app/services/BookService.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
import { Book } from '../models/Book';

@Injectable()
export class BookService {

  url = environment.baseUrl + "/api/book";
  headers = { 'headers': { 'Content-Type': 'application/json', 'Accept': 'application/json' } };

  constructor(private http: HttpClient) { }

  getAllBooks(): Observable<Book[]> {
    return this.http.get<Book[]>(this.url + "/get", this.headers);
  }

  getBookById(id: number): Observable<Book> {
    return this.http.get<Book>(this.url + "/get/" + id, this.headers);
  }

  deleteBookById(id: number): Observable<number> {
    return this.http.delete<number>(this.url + "/delete/" + id, this.headers);
  }

  createBook(book: Book): Observable<number> {
    return this.http.post<number>(this.url + "/create", book, this.headers);
  }

  updateBook(book: Book): Observable<number> {
    return this.http.put<number>(this.url + "/update", book, this.headers);
  }

}

File: backend/src/main/java/com/stackroute/services/BookService.java
package com.stackroute.services;

import java.util.List;
import java.util.Map;

import com.stackroute.models.Book;

public interface BookService {

	public List<Book> getAllBooks(String sortBy, String order);

	public Book getBookById(int id);

	public Integer createBook(Book book);

	public Integer updateBook(Book book);

	public Integer deleteBookById(int id);

	public List<Book> searchBooksByName(String name);

	public Integer patchBook(Integer id, Map<String, Object> map);

}
